jobs/generation/Utilities.groovy

```diff
@@ -339,7 +801,21 @@ class Utilities {
   def static applyPatch(def job, String patch, String projectDir = "") {
     job.with {
       steps {
-        shell("if [ -f ${patch} ]; then git -C \${WORKSPACE}/${projectDir} apply \${WORKSPACE}/${patch}; fi")
+        shell("if [ -f \${WORKSPACE}/${patch} ]; then git -C \${WORKSPACE}/${projectDir} apply \${WORKSPACE}/${patch}; fi")
+      }
+    }
+  }
+
+  /**
+   * Change ownership to jenkins UID of the projectDir
+   *
+   * @param job Job to fix permissions
+   * @param projectDir (optional) Project directory
+   */
+  def static fixPermissions(def job, String projectDir = ".") {
+    job.with {
+      steps {
+        shell("sudo chown \$( id -u \${USER} ):\$( id -u \${USER} ) ${projectDir} -R")
```

Comment:
Does it have sudo rights?

Reply:
Yes, jenkins user has permissions for it. We need sudo here because there are files that created under root in docker (in MS scripts). As jenkins works inside docker I think it isn't security issue.